### PR TITLE
Add Count class for row counting with test and new PredicateSelection…

### PR DESCRIPTION
**Pull request by lemonboston**
…Matcher. Update jems to 1.9. #101

### CODE REVIEW

File: contentpal/src/main/java/org/dmfs/android/contentpal/tools/Count.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+
+    @Override
+    public Integer value()
+    {
+        try
+        {
+            Cursor cursor = mView.rows(EmptyUriParams.INSTANCE, mPredicate, Absent.<String>absent());
+            int count = cursor.getCount();
+            cursor.close();
```
Comment (dmfs) [inline contentpal/src/main/java/org/dmfs/android/contentpal/tools/Count.java]: `cursor.close()` should go into `finally`. Also make sure you have a unit test which covers this case (i.e. throw an exception in `mView.rows(…)` or `cursor.getCount()`).

File: contentpal/src/test/java/org/dmfs/android/contentpal/tools/CountTest.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    @Test
+    public void test_thatCursorGetCountIsReturned() throws RemoteException
+    {
+        View<Object> mockView = mock(View.class, new FailAnswer());
+        Predicate dummyPredicate = mock(Predicate.class, new FailAnswer());
+        Cursor mockCursor = mock(Cursor.class, new FailAnswer());
+
+        doReturn(mockCursor).when(mockView).rows(EmptyUriParams.INSTANCE, dummyPredicate, Absent.<String>absent());
+        doReturn(4).when(mockCursor).getCount();
+        doNothing().when(mockCursor).close();
```
Comment (dmfs) [inline contentpal/src/test/java/org/dmfs/android/contentpal/tools/CountTest.java]: This is probably one of the rare cases in which we actually should use `verify` to ensure `close()` is always called on the cursor. Unless you have a better idea to check that.

File: contentpal/src/main/java/org/dmfs/android/contentpal/tools/Count.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        this(view, new AnyOf());
+    }
+
+
+    @Override
+    public Integer value()
+    {
+        try
+        {
+            Cursor cursor = mView.rows(EmptyUriParams.INSTANCE, mPredicate, Absent.<String>absent());
```
Comment (dmfs) [inline contentpal/src/main/java/org/dmfs/android/contentpal/tools/Count.java]: I'm not sure if it makes a big difference, but maybe rebase and use the `withProjection` method to remove any projection (use `BaseColumns._ID` as the only projection value).
Also, please rename this class to `RowCount` as there is another more efficient way to get the row count which can be used for Content Providers which are backed by an SQL table. I'll create a separate issue for that.

**Review by dmfs — CHANGES_REQUESTED** [review]

**Comment by lemonboston** [comment]
Made all the changes as suggested, ready for review again.

**Review by dmfs — APPROVED** [review]

**Pull request merged**